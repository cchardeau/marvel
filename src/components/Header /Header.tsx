import React from 'react'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import Button from '@material-ui/core/Button'
import Grid from '@material-ui/core/Grid'

const useStyles = makeStyles(theme => ({
  headerContent: {
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(8, 0, 6)
  },
  headerButton: {
    marginTop: theme.spacing(4)
  }
}))

const Header: React.FC = () => {
  // styling
  const classes = useStyles()

  return (
    <div className={classes.headerContent}>
      <Container maxWidth="sm">
        <Typography component="h1" variant="h2" align="center" color="textPrimary" gutterBottom>
          Marvel app
        </Typography>
        <Typography variant="h5" align="center" color="textSecondary" paragraph>
          Whatever happens tomorrow you must promise me one thing. That you will stay who you are. Not a perfect soldier, but a good man.
        </Typography>
        <Typography variant="h5" align="center" color="textSecondary" paragraph>
          <i>Captain America</i>
        </Typography>
        <div className={classes.headerButton}>
          <Grid container spacing={2} justify="center">
            <Grid item>
              <Button variant="contained" color="primary" onClick={(e) => { e.preventDefault(); window.open('https://developer.marvel.com/docs') }}>
                Visit API website
              </Button>
            </Grid>
          </Grid>
        </div>
      </Container>
    </div>
  )
}

export default Header