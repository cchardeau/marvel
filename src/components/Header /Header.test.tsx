import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Header from './Header'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Header />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
  const output = renderer.create(<Header />).toJSON()
  expect(output).toMatchSnapshot()
})
