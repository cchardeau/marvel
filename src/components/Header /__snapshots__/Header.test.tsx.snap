// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-headerContent-174"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <h1
      className="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      Marvel app
    </h1>
    <p
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
    >
      Whatever happens tomorrow you must promise me one thing. That you will stay who you are. Not a perfect soldier, but a good man.
    </p>
    <p
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
    >
      <i>
        Captain America
      </i>
    </p>
    <div
      className="makeStyles-headerButton-175"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Visit API website
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
